ElectronClient/app/gui/NoteText.jsx

```diff
@@ -897,6 +884,41 @@ class NoteTextComponent extends React.Component {
 		});
 	}
 
+	commandSavePdf() {
+		const path = bridge().showSaveDialog({
+			filters: [{ name: _('PDF File'), extensions: ['pdf']}],
+			defaultPath: safeFilename(this.state.note.title),
+		});
+
+		if (path) {
+			// temporarily change the html in the webview to show the note title
+			const newHtml = this.insertHtmlHeading_(this.lastSetHtml_, this.state.note.title);
+			this.webview_.send('setHtml', newHtml);
+
+			this.webview_.printToPDF({}, (error, data) => {
+				if (error) {
+					bridge().showErrorMessageBox(error.message);
+				} else {
+					shim.fsDriver().writeFile(path, data, 'buffer');
+				}
+
+				// refresh the webview contents, undoing our temporary change
+				this.lastSetHtml_ = '';
+				this.render()
+			});
+		}
+	}
+
+	insertHtmlHeading_(s, heading) {
+		const tag = 'h2';
+		let splitStyle = s.split('</style>');
+		const index = splitStyle.length > 1 ? 1 : 0;
+		let toInsert = heading.replace(/&/g,'&amp;').replace(/</g,'&lt;').replace(/>/g,'&gt;');
```

Comment:
I was sure there was already a function to escape HTML somewhere but looks like there isn't. Any chance you could add an escapeHtml function to string-utils and use it here? Something like this for instance https://stackoverflow.com/a/4835406/561309

Reply:
I had looked, and the only one was buried within React and not exposed. Done.